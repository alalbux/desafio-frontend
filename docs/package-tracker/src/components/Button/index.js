import React from 'react';
import Button from '@mui/material/Button';

function StyledButton({ action, children }) {
  return (
    <Button sx={styles.button} onClick={action}>
      {children}
    </Button>
  );
}

const styles = {
  button: {
    backgroundColor: 'white',
    color: 'white',
    textTransform: 'none',
  },
};

export default StyledButton;
