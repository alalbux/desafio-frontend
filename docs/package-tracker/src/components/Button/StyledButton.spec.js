import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Button from '@/components/Button';

describe('Button', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });

  it('calls the provided action function when clicked', () => {
    const action = jest.fn();
    const buttonText = 'Click me';

    render(<Button action={action}>{buttonText}</Button>);

    const buttonElement = screen.getByRole('button');

    fireEvent.click(buttonElement);

    expect(action).toHaveBeenCalled();
  });
});
