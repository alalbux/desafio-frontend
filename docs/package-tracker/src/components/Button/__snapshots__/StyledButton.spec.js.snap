// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-9pht23-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
